inference-engine/src/vpu/graph_transformer/src/middleend/special_stage_processor.cpp

```diff
@@ -5,99 +5,249 @@
 #include "vpu/middleend/special_stage_processor.hpp"
 
 #include <vector>
-#include <set>
 #include <utility>
 
 namespace vpu {
 
-void SpecialStageProcessor::processSplit(
-        const Model& model,
-        const Stage& stage) {
-    IE_ASSERT(stage->type() == StageType::Split);
+namespace {
 
-    auto input = stage->input(0);
+struct NeedCopyDesc {
+    bool isCopyNeed = false;
+    vpu::Optional<bool> isCopyOptional;
```

Comment:
Optional is necessary semantically, because it is really optional :) It is used in `insertCopyOfInput` to setup attribute `optional` for copy stage in some cases.

Reply:
I replaced optional with built-in bool.